rust/src/database/verkle/variants/managed/nodes/mod.rs

```diff
@@ -186,6 +199,36 @@ impl NodeSize for NodeType {
     }
 }
 
+/// Creates the smallest leaf node capable of storing `n` values, initialized with the given
+/// `stem`, `values` and `commitment`.
+pub fn make_smallest_leaf_node_for(
+    n: usize,
+    stem: [u8; 31],
+    values: &[ValueWithIndex],
+    commitment: VerkleCommitment,
+) -> BTResult<Node, Error> {
+    match Node::smallest_leaf_type_for(n) {
+        NodeType::Leaf2 => Ok(Node::Leaf2(Box::new(SparseLeafNode::<2>::from_existing(
+            stem, values, commitment,
+        )?))),
+        NodeType::Leaf256 => {
+            let mut new_leaf = FullLeafNode {
+                stem,
+                commitment,
+                ..Default::default()
+            };
+            for v in values {
+                new_leaf.values[v.index as usize] = v.value;
+            }
+            Ok(Node::Leaf256(Box::new(new_leaf)))
+        }
+        NodeType::Empty | NodeType::Inner => Err(Error::CorruptedState(
+            "received non-leaf type in make_smallest_leaf_node_for".to_owned(),
+        )
```

Comment:
Empty is a leaf, so maybe use a different message for it.
Also, semantically, it would be correct to construct Empty if it was determined to be the smallest node.